libs/datamodel/core/src/transform/ast_to_dml/db/attributes/relation.rs

```diff
@@ -39,3 +40,43 @@ pub(super) fn validate_relation_field_arity(
         ast_relation_field.span
     ));
 }
+
+/// Validates usage of `onUpdate` with the `referentialIntegrity` set to
+/// `prisma`.
+///
+/// This is temporary to the point until Query Engine supports `onUpdate`
+/// actions on emulations.
+pub(super) fn validate_on_update_without_foreign_keys(
+    model_id: ast::ModelId,
+    field_id: ast::FieldId,
+    relation_field: &RelationField<'_>,
+    ctx: &mut Context<'_>,
+) {
+    let referential_integrity = ctx
+        .db
+        .datasource()
+        .map(|ds| ds.referential_integrity())
+        .unwrap_or_default();
+
+    if referential_integrity.uses_foreign_keys() {
+        return;
+    }
+
+    if relation_field
+        .on_update
+        .map(|act| act != ReferentialAction::NoAction)
+        .unwrap_or(false)
```

Comment:
I think this should be `relation_field.on_update.is_some()` instead. Because we want to completely disallow on_update, so that conveys the meaning better and matches with the source-level representation of "on update forbidden".

Reply:
No we do not. We want to see if it's `NoAction` because that works as it says in emulation: does nothing.

Reply:
So we want to allow either `onUpdate: NoAction` or no on update? Ok that also makes sense.